ibis/backends/dask/tests/execution/test_timecontext.py

```diff
@@ -218,3 +231,88 @@ def test_context_adjustment_window_groupby_id(time_table, time_df3):
     # result should adjust time context accordingly
     result = expr.execute(timecontext=context)
     tm.assert_series_equal(result, expected)
+
+
+def test_adjust_context_scope(time_keyed_left, time_keyed_right):
+    """Test that `adjust_context` has access to `scope` by default."""
+
+    @adjust_context.register(CustomAsOfJoin)
+    def adjust_context_custom_asof_join(
+        op: ops.AsOfJoin,
+        timecontext: TimeContext,
+        scope: Optional[Scope] = None,
+    ) -> TimeContext:
+        """Confirms that `scope` is passed in."""
+        assert scope is not None
+        return timecontext
+
+    expr = CustomAsOfJoin(
+        left=time_keyed_left,
+        right=time_keyed_right,
+        predicates='time',
+        by='key',
+        tolerance=ibis.interval(days=4),
+    ).to_expr()
+    expr = expr[time_keyed_left, time_keyed_right.other_value]
+    context = (Timestamp('20170105'), Timestamp('20170111'))
+    expr.execute(timecontext=context)
+
+
+def test_adjust_context_complete_shift(
```

Comment:
This is the new Dask test that is specifically for this bug fix.

Reply:
can you put a refernce to this PR in a comment